asdf/generic_io.py

```diff
@@ -402,7 +403,8 @@ def close(self):
         """
         if self._close:
             self._fd.close()
-            self._fix_permissions()
+            if hasattr(self, "_fix_permissions"):
```

Comment:
This is a minor bugfix that I found when allowing fsspec to natively handle http urls. The `_fix_permissions` method is only present in the RealFile subclass.